From e4ac5deba9287ea6b66b7b9c1f4c4d2ee1cc69b4 Mon Sep 17 00:00:00 2001
From: Simon Munton <simon-hal@munton.demon.co.uk>
Date: Sun, 24 May 2009 16:01:51 +0000
Subject: fixed segfault in hal-storage-mount

Fixed segfault in hal-storage-mount and don't lose fstype in
probe-volume.
---
diff --git a/hald/linux/probing/probe-volume.c b/hald/linux/probing/probe-volume.c
index 882de88..13c775a 100644
--- a/hald/linux/probing/probe-volume.c
+++ b/hald/linux/probing/probe-volume.c
@@ -97,7 +97,7 @@ set_blkid_values (LibHalChangeSet *cs, blkid_probe pr)
 
 	if (blkid_probe_lookup_value(pr, "TYPE", &type, NULL))
 		type = "";
-	if (libhal_changeset_set_property_string (cs, "volume.fstype", type))
+	if (!libhal_changeset_set_property_string (cs, "volume.fstype", type))
 		libhal_changeset_set_property_string (cs, "volume.fstype", "");
 	HAL_DEBUG(("volume.fstype = '%s'", type));
 
--- hal-0.5.11/tools/hal-storage-mount.c	2009-05-31 17:18:32.000000000 +0200
+++ hal-0.5.11.new/tools/hal-storage-mount.c	2009-05-31 17:33:48.000000000 +0200
@@ -719,7 +719,8 @@
 		/* don't consider uid= on vfat, iso9660, hfs and udf change-uid for the purpose of policy
 		 * (since these doesn't contain uid/gid bits) 
 		 */
-		if (strcmp (libhal_volume_get_fstype (volume), "vfat") != 0 &&
+               if (libhal_volume_get_fstype (volume) != NULL &&
+                   strcmp (libhal_volume_get_fstype (volume), "vfat") != 0 &&
 		    strcmp (libhal_volume_get_fstype (volume), "iso9660") != 0 &&
 		    strcmp (libhal_volume_get_fstype (volume), "hfs") != 0 &&
 		    strcmp (libhal_volume_get_fstype (volume), "udf") != 0) {
